### PR TITLE
[JBJCA-1258] Remove driver cache and the interaction with DriverManager

### PR DESCRIPTION
**Pull request by gaol**
Jira: https://issues.jboss.org/browse/JBJCA-1258

1. Removed driverCache from LocalManagedConnectionFactory
2. Removed interaction with DriverManager
3. Removed some unused imports and private methods

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Can one of the admins verify this patch?

**Comment by jesperpedersen** [comment]
ok to test

File: adapters/src/main/java/org/jboss/jca/adapters/jdbc/local/LocalManagedConnectionFactory.java [hunk 1]
```diff
@@ -511,19 +506,6 @@ else if (mc.getProperties().equals(newProps))
    }
 
    /**
-    * Is the properties equal
```
Comment (jesperpedersen) [inline adapters/src/main/java/org/jboss/jca/adapters/jdbc/local/LocalManagedConnectionFactory.java]: Lets leave this method in for now - ideally it should be used in equals(). But that requires more extensive testing

**Comment by jesperpedersen** [comment]
* Add a test to H2DriverTestCase that uses the DriverManager directly to get a connection, and verify that is possible.
* Fix my comment, squash, and rebase.

Otherwise looks good

**Comment by gaol** [comment]
@jesperpedersen I kept the `isEqual(Properties other)` method, and added unit test. please review it again, thank you. :-)

**Comment by jesperpedersen** [comment]
@gaol Looks like there are missing c.close() method calls in there. Could you verify the other JDBC test cases for similar issues ?

**Comment by gaol** [comment]
@jesperpedersen added the missing c.closed() method calls in JDBC test cases. please review it again. Thank you. :-)

**Comment by jesperpedersen** [comment]
Merged

**Pull request merged**